### ISSUE TITLE
Reader: Cards can use some accessibility improvements.

### ISSUE THREAD

**Issue by kwight** [Automattic/wp-calypso#46648]
Reader entries have some accessibility issues like:

- Uses an H1 for the title, when it should be lower
- The author is associated with the previous title

Brought up at https://conf.a11yto.com/ and reported by @jeryj

**Comment by jeryj** [comment]
Quick note that "author is associated with he previous title" is still not entirely fixed by #46651. It's great that they're now semantically contained within the same `<article>`, but ideally the fix is done via restructuring the HTML to have the author content come after the title, rather than needing to go backwards to access the author info.

```
<article>
  <h2>Title</h2>
  <p>Author info</p>
  <time datetime="[timestamp]">October, 22</time>
  <p>Article summary</p>
</article>
```
